hadoop-hdfs-project/hadoop-hdfs-httpfs/src/main/java/org/apache/hadoop/fs/http/server/HttpFSServerWebApp.java

```diff
@@ -110,9 +116,23 @@ public void init() throws ServerException {
   @Override
   public void destroy() {
     SERVER = null;
+    if (metrics != null) {
+      metrics.shutdown();
+    }
     super.destroy();
   }
 
+  public static void setMetrics(Configuration config) {
+    LOG.info("Initializing HttpFSServerMetrics");
+    HttpFSServerWebApp.metrics =
```

Comment:
Can't you just use metrics instead of HttpFSServerWebApp.metrics?  Same a few lines below.